src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.FixAllProvider.cs

```diff
@@ -0,0 +1,80 @@
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.EnableNullable
+{
+    internal partial class EnableNullableCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        internal sealed override CodeAnalysis.CodeRefactorings.FixAllProvider? GetFixAllProvider()
+            => FixAllProvider.Instance;
+
+        private sealed class FixAllProvider : CodeAnalysis.CodeRefactorings.FixAllProvider
+        {
+            public static readonly FixAllProvider Instance = new();
+
+            private FixAllProvider()
+            {
+            }
+
+            public override IEnumerable<FixAllScope> GetSupportedFixAllScopes()
+                => ImmutableArray.Create(FixAllScope.Solution);
+
+            public override Task<CodeAction?> GetFixAsync(FixAllContext fixAllContext)
+            {
+                Debug.Assert(fixAllContext.Scope == FixAllScope.Solution);
+                return Task.FromResult<CodeAction?>(new FixAllCodeAction(EnableNullableReferenceTypesInSolutionAsync));
+
+                async Task<Solution> EnableNullableReferenceTypesInSolutionAsync(CodeActionPurpose purpose, CancellationToken cancellationToken)
+                {
+                    var solution = fixAllContext.Solution;
+                    foreach (var projectId in solution.ProjectIds)
+                    {
+                        var project = solution.GetRequiredProject(projectId);
+                        if (!ShouldOfferRefactoring(project))
+                            continue;
+
+                        solution = await EnableNullableReferenceTypesAsync(project, purpose,
+                            fixAllContext.GetOptionsProvider(), fixAllContext.CancellationToken).ConfigureAwait(false);
+                    }
+
+                    return solution;
+                }
+            }
+
+            private sealed class FixAllCodeAction : CodeAction.SolutionChangeAction
+            {
+                private readonly Func<CodeActionPurpose, CancellationToken, Task<Solution>> _createChangedSolution;
+
+                public FixAllCodeAction(Func<CodeActionPurpose, CancellationToken, Task<Solution>> createChangedSolution)
+                : base(
+                    CSharpFeaturesResources.Enable_nullable_reference_types_in_solution,
+                    cancellationToken => createChangedSolution(CodeActionPurpose.Apply, cancellationToken),
+                    nameof(CSharpFeaturesResources.Enable_nullable_reference_types_in_solution))
+                {
+                    _createChangedSolution = createChangedSolution;
+                }
+
+                protected override async Task<IEnumerable<CodeActionOperation>> ComputePreviewOperationsAsync(CancellationToken cancellationToken)
+                {
+                    var changedSolution = await _createChangedSolution(CodeActionPurpose.Preview, cancellationToken).ConfigureAwait(false);
+                    if (changedSolution is null)
```

Comment:
Is this reachable?

Reply:
I have kept this consistent with the code action registered for the original refactoring: https://github.com/dotnet/roslyn/blob/d1427b1a828a4a4a83a2bd6b2fce6c5e3eff674c/src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs#L255-L276